src/Chat.cs

```diff
@@ -83,7 +164,7 @@ public IAsyncEnumerable<string> SendAsync(string message, IEnumerable<string> im
 	/// <param name="tools">Tools that the model can make use of, see https://ollama.com/blog/tool-support. By using tools, response streaming is automatically turned off</param>
 	/// <param name="imagesAsBase64">Base64 encoded images to send to the model</param>
 	/// <param name="cancellationToken">The token to cancel the operation with</param>
-	public IAsyncEnumerable<string> SendAsync(string message, IEnumerable<Tool>? tools, IEnumerable<string>? imagesAsBase64 = default, CancellationToken cancellationToken = default)
+	public IAsyncEnumerable<string> SendAsync(string message, IReadOnlyCollection<Tool>? tools, IEnumerable<string>? imagesAsBase64 = default, CancellationToken cancellationToken = default)
```

Comment:
To avoid multiple enumerations over an IEnumerable due to the internal logic around tools in the method, I converted the declaration to be a `IReadOnlyCollection` instead of `IEnumerable`. 

This is a small, but potentially breaking change.